src/crimson/osd/heartbeat.cc

```diff
@@ -310,19 +310,21 @@ seastar::future<> Heartbeat::maybe_share_osdmap(
   }
   auto& peer = found->second;
 
-  if (m->map_epoch > peer.get_projected_epoch()) {
-    logger().debug("{} updating peer {} session's projected_epoch"
-                   "from {} to ping map epoch of {}",
-                   __func__, from, peer.get_projected_epoch(),
-                   m->map_epoch);
-    peer.set_projected_epoch(m->map_epoch);
+  // Peer already has newer projeceted osdmap epoch,
+  // no new incremental maps to share.
+  if (current_osdmap_epoch <= peer.get_projected_epoch()) {
```

Comment:
For example, in this branch, suppose that current_osdmap_epoch is 10, peer.get_projected_epoch() is 20, and m->map_epoch is 30.  If we skip updating projected_epoch, we'll try to send the peer a map once current_osdmap_epoch > 20, when we actually don't need to send one until it passes 30.